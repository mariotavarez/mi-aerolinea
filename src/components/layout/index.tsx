// Navbar
export { Navbar } from "./navbar/Navbar";
// Main
export { Main } from "./main/Main";
// Footer
export { Footer } from "./footer/Footer";
