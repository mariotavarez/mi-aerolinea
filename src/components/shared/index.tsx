// Button
export { Button } from "./button/Button";
// Search
export { Search } from "./search/Search";
// Cards
export {
  CardAeroline,
  CardAerolineDetail,
  CardAerolinePrice,
  CardPrice,
} from "./cards/";
// Paragraph
export { Paragraph } from "./paragraph/Paragraph";
// Screen
export { Screen } from "./screen/Screen";
// Form Generator
export { FormGenerator } from "./forms/FormGenerator";
