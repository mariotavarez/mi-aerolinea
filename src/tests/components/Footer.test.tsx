// Enzyme
import { shallow } from "enzyme";
// Layout
import { Footer } from "../../components/layout";

describe("Pruebas en componente <Footer />", () => {
  let wrapper = shallow(<Footer />);
  // Renderizado de componente
  test("debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  //   Siguenos en
  test("la seccion follow us debe mostrar el siguiente texto: Siguenos en", () => {
    const followUs = wrapper.find("p").first().text();
    expect(followUs).toBe("Siguenos en");
  });
  //   Derechos reservados
  test("la seccion copyright debe mostrar leyenda de derechos reservados", () => {
    const copyright = wrapper.find("p").at(1).text();
    // Get Full Year
    const year = new Date().getFullYear();
    expect(copyright).toBe(
      `Copyright ${year} Â© MiAerolinea.com. Todos los derechos reservados.`
    );
  });
});
