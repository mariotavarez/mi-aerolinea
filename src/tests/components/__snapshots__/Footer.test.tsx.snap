// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Footer /> debe mostrar el componente correctamente 1`] = `
<footer
  className="container_footer"
>
  <div
    className="follow_us"
  >
    <p>
      Siguenos en
    </p>
  </div>
  <div
    className="social_media"
  >
    <span
      key="0"
    >
      <FaInstagram
        size="2rem"
      />
    </span>
    <span
      key="1"
    >
      <FaFacebookSquare
        size="2rem"
      />
    </span>
    <span
      key="2"
    >
      <FaTwitter
        size="2rem"
      />
    </span>
  </div>
  <div
    className="copyright"
  >
    <p>
      Copyright 
      2022
       Â© MiAerolinea.com. Todos los derechos reservados.
    </p>
  </div>
</footer>
`;
